Desktop/components/JASP/Widgets/FileMenu/Database.qml

```diff
@@ -0,0 +1,379 @@
+                height:				dbHostnameItem.height
+
+                Text
+                {
+                    id:						dbNameLabel
+                    text:					qsTr("Name")
+                    width:					dbHostnameLabel.width
+                    anchors.verticalCenter: parent.verticalCenter
+                }
+
+                PrefsTextInput
+                {
+                    id:				dbNameInput
+					nextEl:			connectButton
+                    text:			fileMenuModel.database.database
+                    onTextChanged:	fileMenuModel.database.database = text;
+
+                    x:				dbHostnameLabel.width
+					width:			dbHostnameInput.width - (!browseDbButton.enabled ? 0 : browseDbButton.width + jaspTheme.generalAnchorMargin)
+                }
+
+				RoundedButton
+				{
+					id:					browseDbButton
+					text:				qsTr("Browse")
+					KeyNavigation.tab:	dbUsernameInput.textInput
+					enabled:			fileMenuModel.database.dbMaybeFile
+					visible:			enabled
+					onClicked:			fileMenuModel.database.browseDbFile();
+					anchors.right:		parent.right
+				}
+            }
+
+            Item
+            {
+                anchors.left:		parent.left;
+                anchors.right:		parent.right;
+                height:				dbHostnameItem.height
+
+                Text
+                {
+                    id:						dbUsernameLabel
+                    text:					qsTr("Username")
+                    width:					dbHostnameLabel.width
+                    anchors.verticalCenter: parent.verticalCenter
+                }
+
+                PrefsTextInput
+                {
+                    id:				dbUsernameInput
+                    nextEl:			dbPasswordInput.textInput
+                    text:			fileMenuModel.database.username
+                    onTextChanged:	fileMenuModel.database.username = text;
+
+                    x:				dbHostnameLabel.width
+                    width:			dbHostnameInput.width
+                }
+            }
+
+            Item
+            {
+                anchors.left:		parent.left;
+                anchors.right:		parent.right;
+                height:				dbHostnameItem.height
+
+                Text
+                {
+                    id:						dbPasswordLabel
+                    text:					qsTr("Password")
+                    width:					dbHostnameLabel.width
+                    anchors.verticalCenter: parent.verticalCenter
+                }
+
+                PrefsTextInput
+                {
+                    id:						dbPasswordInput
+                    nextEl:					connectButton
+                    text:					fileMenuModel.database.password
+                    onTextChanged:			fileMenuModel.database.password = text;
+
+                    textInput.echoMode:		TextInput.Password
+
+                    x:						dbHostnameLabel.width
+                    width:					dbHostnameInput.width
+                }
+            }
+
+            RoundedButton
```

Comment:
This button should appear only when the database is not yet connected. When connected it should show 'Close database', and the connection settings should be disabled.

Reply:
Ok but I think this is a fundamental thing here, the "Open" menu is not for changing the currently loaded dataset at all.
So a database is never "connected" once you enter that screen right?

What would the behaviour even be?
You press "Load into JASP" and then you lose all changes you made? 
Why would you change the connection once it works?